crates/shim/src/asynchronous/util.rs

```diff
@@ -87,6 +87,12 @@ pub async fn write_str_to_file(filename: impl AsRef<Path>, s: impl AsRef<str>) -
     Ok(())
 }
 
+pub async fn read_pid_from_file(pid_path: &Path) -> Result<i32> {
```

Comment:
Question: Should we maintain parity between sync and async APIs? Specifically, should any additions or modifications to one module be reflected in the other?

Reply:
That's right, but `read_pid_from_file` this function has been implemented in `synchronous`.